source/PlayerInfo.cpp

```diff
@@ -2041,6 +2043,55 @@ void PlayerInfo::SetTravelDestination(const Planet *planet)
 
 
 
+void PlayerInfo::QueueTeleport(const Planet *destination, bool flagshipOnly)
+{
+	teleportPlanet = destination;
+	flagshipOnly = flagshipOnly;
+}
+
+
+
+void PlayerInfo::DoQueuedTeleport()
+{
+	if(!teleportPlanet)
+		return;
+	if(!planet)
+	{
+		teleportPlanet = nullptr;
+		return;
+	}
+
+	flagship->SetSystem(teleportPlanet->GetSystem());
```

Comment:
Should we check if the flagship pointer is set before those updates? (I can imagine that some teleports happen while the flagship is not yet set.)

Reply:
I can do it.

Reply:
done

Reply:
I still feel something should be done here, like maybe calling FlagshipPtr(), and checking it's return value.

Reply:
Oh, whoops changing it

Reply:
done

Reply:
a little bit further up